go-controller/pkg/ovn/endpoints.go

```diff
@@ -284,3 +254,20 @@ func (ovn *Controller) clearVIPsAddRejectACL(svc *kapi.Service, lb, ip string, p
 		klog.Errorf("Error in clearing endpoints for lb %s: %v", lb, err)
 	}
 }
+
+// hasHostEndpoints determines if a slice of endpoints contains a host networked pod
+func hasHostEndpoints(endpointIPs []string) bool {
```

Comment:
I think this always return true https://play.golang.org/p/CEybe3YeFWt

Reply:
```suggestion
func hasHostEndpoints(endpointIPs []string) bool {
	for _, endpointIP := range endpointIPs {
		for _, clusterNet := range config.Default.ClusterSubnets {
			if clusterNet.CIDR.Contains(net.ParseIP(endpointIP)) {
				return true
			}
		}

	}
	return false
```

Reply:
the cluster subnets are the pod ip subnets. So any endpoint ip is not found in the cluster subnets, then we consider it as a host network pod:

https://play.golang.org/p/MTlrL6EqcJm

Reply:
heh, I got it completely wrong then

Reply:
just switching true and false in the function you wrote wont work, because it will return after the first endpoint IP is evaluated and found to be in cluster subnet. That means the second IP will not be evaluated which may have been a host networked pod.